apisix/plugins/redirect.lua

```diff
@@ -55,13 +55,32 @@ local schema = {
         },
         http_to_https = {type = "boolean"},
         encode_uri = {type = "boolean", default = false},
-        append_query_string = {type = "boolean", default = false},
+        append_query_string = {type = "boolean"},
+    },
+    allOf = {
+        {
+            oneOf = {
+                {required = {"uri"}},
+                {required = {"regex_uri"}},
+                {required = {"http_to_https"}},
+            },
+        },
+        -- append_query_string and http_to_https are exclusive
+        {
+            oneOf = {
+                {required = {"append_query_string"}},
+                {required = {"http_to_https"}},
+                {
+                    ["not"] = {
+                        anyOf = {
+                            {required = {"append_query_string"}},
+                            {required = {"http_to_https"}},
+                        },
+                    },
+                },
+            },
+        },
     },
-    oneOf = {
-        {required = {"uri"}},
-        {required = {"regex_uri"}},
-        {required = {"http_to_https"}}
```

Comment:
https://github.com/apache/apisix/blob/ccfb00055c9814f3a94514e48aa583700ab94189/apisix/schema_def.lua#L420
We can use `not` operation to show the exclusive relationship?

Reply:
I have tried, but not succeed
```
["not"] = {
    allOf = {
        {required = {"append_query_string"}},
        {required = {"http_to_https"}},
    }
}  
```
and

```
["not"] = {required = {"append_query_string", "http_to_https"}},
```

Reply:
```
http_to_https = {
    ["not"] = {required = {"uri", "regex_uri", "append_query_string"}}
},

uri = {
    ["not"] = {required = {"regex_uri", "http_to_https"}}
},

regex_uri = {
    ["not"] = {required = {"uri", "http_to_https"}}
},
```

what about this form?

Reply:
Give it a try:
```
    oneOf = {
        {required = {"uri"}},
        {required = {"regex_uri"}},
        {required = {"http_to_https"}, ["not"] = {
           required = {"append_query_string"}
        }}
    }
```

Reply:
> Give it a try:
> 
> ```
>     oneOf = {
>         {required = {"uri"}},
>         {required = {"regex_uri"}},
>         {required = {"http_to_https"}, ["not"] = {
>            required = {"append_query_string"}
>         }}
>     }
> ```

It does work,  i have updated, please check @spacewander